src/serialize.rs

```diff
@@ -623,6 +627,40 @@ impl Deserialize for Vec<u8> {
     }
 }
 
+
+pub struct StringReader {
+    bytes_reader: BytesReader,
+}
+
+impl TypeRead for StringReader {
```

Comment:
Do these impls truncate trailing nulls?  It feels to me like they should.

Reply:
Ah!  I think it does, so long as the `|S` dtype is used in the file. (I had forgotten that the `S` dtype exists!)

Keeping this open as a reminder that a test is needed though.